File: kaleido-gradle-plugin/src/main/java/com/tongsr/kaleido/gradle/dsl/KaleidoSeed.java
package com.tongsr.kaleido.gradle.dsl;

import java.util.Objects;
import org.gradle.api.provider.Provider;

public final class KaleidoSeed {
    private Provider<String> provider;

    public void set(Provider<String> provider) {
        this.provider = Objects.requireNonNull(provider, "provider");
    }

    public boolean isConfigured() {
        return provider != null;
    }

    public Provider<String> getProvider() {
        if (provider == null) {
            throw new IllegalStateException("No explicit Kaleido seed Provider is configured");
        }
        return provider;
    }
}

File: kaleido-gradle-plugin/src/main/kotlin/com/tongsr/kaleido/gradle/dsl/KaleidoSeed.kt
package com.tongsr.kaleido.gradle.dsl

import org.gradle.api.provider.Provider
import java.util.Objects

class KaleidoSeed {
    private var configuredProvider: Provider<String>? = null

    fun set(provider: Provider<String>) {
        configuredProvider = Objects.requireNonNull(provider, "provider")
    }

    fun isConfigured(): Boolean = configuredProvider != null

    fun getProvider(): Provider<String> = configuredProvider
        ?: throw IllegalStateException("No explicit Kaleido seed Provider is configured")
}
